src/data.ts

```diff
@@ -161,7 +161,7 @@ export function getFirst(query: any | string | RegExp) : Promise<IDataType> {
     return getFirstByName(<string>query);
   }
   query.limit = 1;
-  return list(query).then((result) => {
+  return list(query).then<IDataType>((result) => {
```

Comment:
Thanks. The other problem I quoted above still remains... I'll see if I can understand it better this morning.